test/common/stats/lazyinit_stats_test.cc

```diff
@@ -27,13 +27,29 @@ class LazyInitStatsTest : public testing::Test {
   AwesomeStatNames stats_names_{symbol_table_};
 };
 
-using LazyAwesomeStats = LazyCompatibleStats<AwesomeStats>;
+using MyStats = LazyCompatibleStats<AwesomeStats>;
+
+TEST_F(LazyInitStatsTest, NonLazyNoInitializedGauge) {
+  {
+    ScopeSharedPtr scope = store_.createScope("bluh");
+    MyStats non_lazy_y = MyStats::create(scope, stats_names_, false);
+    EXPECT_EQ(TestUtility::findGauge(store_, "bluh.AwesomeStats.initialized"), nullptr);
+    non_lazy_y->foo_.inc();
+    MyStats lazy_x = MyStats::create(scope, stats_names_, true);
```

Comment:
nit: from line 38 you are testing lazy stats so just split into a separate method not named "NonLazy*" :)

You can also put the scope in your test fixture instead of repeating it as the first line of every test.

Reply:
removed.